### PR TITLE
Update elastic-agent-system-metrics to v0.8.1

### PR DESCRIPTION
**Pull request by belimawr**
## Proposed commit message

Update elastic-agent-system-metrics to v0.8.1 to enable collecting
memory and CPU metrics from privileged process on Windows.

Fix python test to ensure the `cmdLine` is found in at least one
process instead of them all because we cannot fetch the `cmdLine` from
privileged process on Windows.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist
- [x] Update `elastic-agent-system-metrics` to the fixed version (PR: https://github.com/elastic/elastic-agent-system-metrics/pull/107)

## How to test this PR locally
1. Install Elastic-Agent with Elastic-Defend integration enabled
2. Run Metricbeat (from this PR) as administrator with process metrics enabled
3. Ensure events from `elastic-endpoint.exe` contain CPU and memory metrics.

## Related issues

- Closes https://github.com/elastic/beats/issues/17314

~~## Use cases~~
## Screenshots
Screenshot 2023-11-02 150329

~~## Logs~~

### CODE REVIEW

**Review by cmacknz — APPROVED** [review]

**Review by fearful-symmetry — APPROVED** [review]

File: metricbeat/module/system/test_system.py
Comment (belimawr) [inline metricbeat/module/system/test_system.py:443]: @fearful-symmetry, could you do a sanity check on this last change? I believe it is ok to change, but I'd like a second opinion before merging.

Comment (fearful-symmetry) [reply metricbeat/module/system/test_system.py:443]: It looks like the `assertTrue()` line was just moved out of the loop block to check for the cmdline globally? That seems fine, but wouldn't it require the last event to set `found_cmdline |= "cmdline" in process` in order to pass? Maybe a counter that needs to be > 1 would work?

Comment (belimawr) [reply metricbeat/module/system/test_system.py:443]: > but wouldn't it require the last event to set found_cmdline |= "cmdline" in process in order to pass?

Comment (belimawr) [reply metricbeat/module/system/test_system.py:443]: > It looks like the assertTrue() line was just moved out of the loop block to check for the cmdline globally?

Yes, that was my intent because it already does an `OR` operation on itself. Being honest I was very confused by the `assertTrue()` being inside the loop and having `found_cmdline |= "cmdline" in process`.

https://github.com/elastic/beats/blob/bab5bb460c985cf0fa129759e3c891bf7f9ff65d/metricbeat/module/system/test_system.py#L421-L427

> but wouldn't it require the last event to set found_cmdline |= "cmdline" in process in order to pass?

That's an `OR` operator, so as long as `"cmdline" in process` returns `True` at least once, then `found_cmdline` will always be `True`.

**Pull request merged**
